tests/tests_api/test_group.py

```diff
@@ -77,24 +87,75 @@ def test_public_group_retrieve(api_client):
         "coordinators": [
             {
-                "id": coordinator.id,
+                "id": coordinator.pk,
                 "full_name": coordinator.personal_info.full_name,
             }
             for coordinator in group.coordinators.all()
         ],
         "teachers": [
             {
-                "id": teacher.id,
+                "id": teacher.pk,
                 "full_name": teacher.personal_info.full_name,
             }
             for teacher in group.teachers.all()
         ],
         "students": [
             {
-                "id": student.id,
+                "id": student.pk,
                 "full_name": student.personal_info.full_name,
             }
             for student in group.students.all()
         ],
         "is_for_staff_only": group.is_for_staff_only,
     }
+
+
+class TestPublicGroupStart:
+    def test_public_group_start_general_check(self, api_client):
+        group = baker.make(Group, _fill_optional=True, make_m2m=True)
+
+        with transaction.atomic():
+            group.status = GroupStatus.AWAITING_START
+            # to make sure the status_since really gets updated:
+            group.status_since = timezone.now() - datetime.timedelta(days=1, hours=1, minutes=10)
+            group.save()
+            group.coordinators.update(status=CoordinatorStatus.WORKING_BELOW_THRESHOLD)
+            group.students.update(status=StudentStatus.AWAITING_START)
+            group.teachers.update(status=TeacherStatus.AWAITING_START)
+
+        response = api_client.post(self._make_url(group))
+
+        assert response.status_code == status.HTTP_201_CREATED
+
+        group = Group.objects.get(pk=group.id)  # reload, or else status will be the old one
+        assert group.status == GroupStatus.WORKING
+
+        common_status_since = group.status_since
+        assert common_status_since.day == timezone.now().day
+        assert common_status_since.hour == timezone.now().hour
+        assert common_status_since.minute == timezone.now().minute
+
+        for coordinator in group.coordinators.iterator():
+            assert coordinator.status in (
+                CoordinatorStatus.WORKING_BELOW_THRESHOLD,
+                CoordinatorStatus.WORKING_OK,
+                CoordinatorStatus.WORKING_LIMIT_REACHED,
+            )
+            assert coordinator.status_since == common_status_since
+
+        for student in group.students.iterator():
+            assert student.status == StudentStatus.STUDYING
+            assert student.status_since == common_status_since
+
+        for teacher in group.teachers.iterator():
+            assert teacher.status in (
+                TeacherStatus.TEACHING_ACCEPTING_MORE,
+                TeacherStatus.TEACHING_NOT_ACCEPTING_MORE,
+            )
+            assert teacher.status_since == common_status_since
+
+        # TODO test creation of log events
+
+    @staticmethod
+    def _make_url(group: Group) -> str:
```

Comment:
There is `reverse()` in django for it :) 
https://docs.djangoproject.com/en/4.2/ref/urlresolvers/#reverse

Reply:
Fixed (I still decided to leave the `_make_url()` method, though